m2m/m2m.py

```diff
@@ -57,28 +49,31 @@ class MetadataRecord(object):
     def __init__(self, metadataCreator, addDate=False):
         # create our initial tree
         self.root_element = PYUNTL_DISPATCH['metadata']()
-        self.map("basic", "meta", metadataCreator, qualifier="metadataCreator")
+        self.mapping("basic", "meta", metadataCreator, qualifier="metadataCreator")
         if addDate is True:
-            self.map("basic", "meta", "%s" % time.strftime(
+            self.mapping("basic", "meta", "%s" % time.strftime(
                      "%Y-%m-%d, %H:%M:%S"), qualifier="metadataCreationDate")
 
-    def __str__(self):
+    def __bytes__(self):
         return untlpydict2xmlstring(untlpy2dict(self.root_element))
 
+    def __str__(self):
+        return self.__bytes__().decode()
+
     def setBaseDirectory(self, BaseDirectory):
         self.baseDirectory = BaseDirectory
 
     def setFolderName(self, FolderName):
         self.foldername = FolderName
 
-    def map(self, elementType, elementName, elementValue, qualifier=None,
+    def mapping(self, elementType, elementName, elementValue, qualifier=None,
             required=True, info="", location="", agent_type="", split="",
             function=None):
         """
         Mapping Function
         """
 
-        if elementType is not "basic" and elementType is not "agent":
+        if elementType not in {"basic", "agent"}:
```

Comment:
Is there a reason to use a set here instead of a list or tuple?

Reply:
nothing particular honestly.

Reply:
I would personally change it to a list or tuple then, though the only real reason I have for it is for consistency with how we typically do this in other repos.